designs/0006-labels.md

```diff
@@ -0,0 +1,289 @@
+
+#### Conclusion
+
+We will add these labels to WellKnownLabels + requirements:
+* kubernetes.azure.com/scalesetpriority
+* kubernetes.azure.com/node-image-version
+* kubernetes.azure.com/fips_enabled
+* kubernetes.azure.com/os-sku
+* kubernetes.azure.com/mode
+
+We will consider adding these labels to WellKnownLabels + requirements in the future:
+
+* kubernetes.azure.com/ebpf-dataplane
+* kubernetes.azure.com/os-sku-effective
+* kubernetes.azure.com/os-sku-requested
+* kubernetes.azure.com/security-type
+* kubernetes.azure.com/accelerator
+
+We will allow these labels to be set on the NodePool/NodeClaim (in addition to the labels we already support as schedulable outlined in the tables above):
+* kubernetes.azure.com/ebpf-dataplane
+* kubernetes.azure.com/cluster-health-monitor-checker-synthetic (due to its high usage...) **TODO:** Learn what this is
+
+Based on usage data, the following `kubernetes.azure.com` labels have some usage today in `NodePool` labels or requirements.
+
+Most usage
+
+* kubernetes.azure.com/ebpf-dataplane
+* kubernetes.azure.com/mode
+* kubernetes.azure.com/cluster-health-monitor-checker-synthetic
+* kubernetes.azure.com/scalesetpriority
+
+Less usage
+* kubernetes.azure.com/agentpool
+* kubernetes.azure.com/accelerator
+* kubernetes.azure.com/agentpool-family
+* kubernetes.azure.com/os-sku
+* kubernetes.azure.com/aksnodeclass
+* kubernetes.azure.com/storageprofile
+* kubernetes.azure.com/storagetier
+
+We already block usage of the `karpenter.azure.com` and `karpenter.sh` namespaces except for certain labels we understand. This will not change.
+
+We will also block all `kubernetes.azure.com` namespaced labels except for those called out above as either well known labels, or allowed.
+
+### Decision 5: How will we get labels that Karpenter doesn't currently write set on nodes?
+
+#### Option A: Wait for Machine API migration
+
+Migrating to the machine API will automatically get all missing labels set on VMs allocated via the machine API.
+
+#### Option B: Add some/all labels to our current label writing mechanism
+
+There are some labels that Karpenter sets but that aren't schedulable, defined at [AddAgentBakerGeneratedLabels](https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/providers/imagefamily/labels/labels.go#L23).
+This has already been accounted for in the tables above. We could add more labels there.
+
+#### Conclusion: Wait for machine API for complete parity
+
+Machine API is not that far away, and we already write most critical labels. Some additional labels such as `kubernetes.azure.com/os-sku` and `kubernetes.azure.com/fips_enabled` will be enabled for scheduling
+(and thus be written to the nodes) based on decision 4, but other labels and full label parity will come with the machine API migration.
+
+## FAQ
+
+### How do the labels get onto the Node?
```

Comment:
The following describes the sequence in the reverse order, which makes sense, but probably worth noting explicitely

Reply:
Added. Also clarified there are two paths here. In the actual issue tracking this there is an item to remove one of the paths (IMO the non-kubelet one), at least for the "core" labels we own. User labels will still be set by Karpenter core via the path documented here.